mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -1085,229 +392,83 @@ func resourceTemplateContractServiceGraphRead(d *schema.ResourceData, m interfac
 
-			tp := strings.Split(siteMap["provider_connector_cluster_interface"].(string), "/")
-			token := strings.Split(tp[len(tp)-1], "-")
-			allMap["provider_connector_cluster_interface"] = token[1]
-
-			tp = strings.Split(siteMap["consumer_connector_cluster_interface"].(string), "/")
-			token = strings.Split(tp[len(tp)-1], "-")
-			allMap["consumer_connector_cluster_interface"] = token[1]
-
-			if siteMap["provider_connector_redirect_policy"] != nil {
-				tp := strings.Split(siteMap["provider_connector_redirect_policy"].(string), "/")
-				token1 := strings.Split(tp[1], "-")
-				allMap["provider_connector_redirect_policy_tenant"] = token1[1]
-
-				token2 := strings.Split(tp[len(tp)-1], "-")
-				allMap["provider_connector_redirect_policy"] = token2[1]
-			}
-			if siteMap["consumer_connector_redirect_policy"] != nil {
-				tp := strings.Split(siteMap["consumer_connector_redirect_policy"].(string), "/")
-				token1 := strings.Split(tp[1], "-")
-				allMap["consumer_connector_redirect_policy_tenant"] = token1[1]
-
-				token2 := strings.Split(tp[len(tp)-1], "-")
-				allMap["consumer_connector_redirect_policy"] = token2[1]
-			}
-
-			if siteMap["provider_subnet_ips"] != nil {
-				allMap["provider_subnet_ips"] = siteMap["provider_subnet_ips"]
-			}
-			if siteMap["consumer_subnet_ips"] != nil {
-				allMap["consumer_subnet_ips"] = siteMap["consumer_subnet_ips"]
-			}
-
 			nodeList = append(nodeList, allMap)
 		}
-		d.Set("schema_id", schemaID)
-		d.Set("site_id", siteID)
-		d.Set("template_name", templateName)
 		d.Set("node_relationship", nodeList)
-		d.Set("contract_name", contractName)
+		d.SetId(fmt.Sprintf("%s/templates/%s/contracts/%s", d.Get("schema_id").(string), templateName, contractName))
 
-		if d.Get("service_graph_name") == serviceGraph {
-			d.SetId(serviceGraph)
-		} else {
-			d.SetId("")
-		}
 	} else {
 		d.SetId("")
 	}
-
-	log.Printf("[DEBUG] Completed Read Template Contract Service Graph")
 	return nil
 }
 
-func resourceTemplateContractServiceGraphDelete(d *schema.ResourceData, m interface{}) error {
-	log.Printf("[DEBUG] Begining Delete Template Contract Service Graph")
+// PostSchemaTemplateContractServiceGraphConfig creates a new service graph configuration for a template contract.
+//
+// Parameters:
+//   - action: The action to perform.
+//   - d: The schema resource data.
+//   - m: The client interface.
+//
+// Returns:
+//   - An error if there was a problem creating the service graph configuration.
+func PostSchemaTemplateContractServiceGraphConfig(action string, d *schema.ResourceData, m interface{}) error {
 	msoClient := m.(*client.Client)
-
-	schemaID := d.Get("schema_id").(string)
-	siteID := d.Get("site_id").(string)
+	schemaId := d.Get("schema_id").(string)
 	TemplateName := d.Get("template_name").(string)
 	contractName := d.Get("contract_name").(string)
+	serviceGraph := d.Get("service_graph_name").(string)
```

Comment:
should be keep the naming aligned with the schema definition?
```suggestion
	serviceGraphName := d.Get("service_graph_name").(string)
```